identifiers/validate.go

```diff
@@ -25,14 +25,16 @@ var (
 	//
 	// Rules for domains, defined in RFC 1035, section 2.3.1, are used for
 	// identifiers.
-	identifierRe = regexp.MustCompile(reAnchor(label + reGroup("[.]"+reGroup(label)) + "*"))
+	identifierRe = regexp.MustCompile(reAnchor(reCapture(label) + reGroup("[.]"+reCapture(label)) + "*"))
 
 	errIdentifierInvalid = errors.Errorf("invalid, must match %v", identifierRe)
+
+	errLengthInvalid = errors.Errorf("invalid, name must less than 255, and label must less than 63.")
```

Comment:
It is because `func IsInvalid(err error) bool` checks root cause, and called outside.

Here emits `errLengthInvalid`, and let package-consumer know if it is error root cause.

Maybe you mean all should use `errIdentifierInvalid`? But how to define the language, or like:

```go
errIdentifierInvalid = errors.Errorf("invalid, must match %v, with label size limitation of 63 and name size limitation of 255", identifierRe)
```